apps/condo/domains/analytics/schema/ExternalReport.test.js

```diff
@@ -0,0 +1,171 @@
+                const [objCreated] = await createTestExternalReport(admin)
+
+                const client = await makeClient()
+                await expectToThrowAuthenticationErrorToObj(async () => {
+                    await updateTestExternalReport(client, objCreated.id, {
+                        title: faker.datatype.string(),
+                    })
+                })
+            })
+        })
+
+        describe('hard delete', () => {
+            test('admin can\'t', async () => {
+                const admin = await makeLoggedInAdminClient()
+                const [objCreated] = await createTestExternalReport(admin)
+
+                await expectToThrowAccessDeniedErrorToObj(async () => {
+                    await ExternalReport.delete(admin, objCreated.id)
+                })
+            })
+
+            test('user can\'t', async () => {
+                const admin = await makeLoggedInAdminClient()
+                const [objCreated] = await createTestExternalReport(admin)
+
+                const client = await makeClientWithNewRegisteredAndLoggedInUser()
+                await expectToThrowAccessDeniedErrorToObj(async () => {
+                    await ExternalReport.delete(client, objCreated.id)
+                })
+            })
+        })
+
+        describe('read', () => {
+            test('admin can', async () => {
+                const admin = await makeLoggedInAdminClient()
+                const [obj, attrs] = await createTestExternalReport(admin)
+
+                const objs = await ExternalReport.getAll(admin, {}, { sortBy: ['updatedAt_DESC'] })
+
+                expect(objs.length).toBeGreaterThanOrEqual(1)
+                expect(objs).toEqual(expect.arrayContaining([
+                    expect.objectContaining({
+                        id: obj.id,
+                        iframeUrl: obj.iframeUrl,
+                        type: obj.type,
+                        title: obj.title,
+                        description: obj.description,
+                        meta: obj.meta,
+                        organization: obj.organization,
+                    }),
+                ]))
+            })
+
+            test('user can', async () => {
+                const admin = await makeLoggedInAdminClient()
+                const [obj, attrs] = await createTestExternalReport(admin)
+
+                const client = await makeClientWithNewRegisteredAndLoggedInUser()
+                const objs = await ExternalReport.getAll(client, {}, { sortBy: ['updatedAt_DESC'] })
+
+                expect(objs.length).toBeGreaterThanOrEqual(1)
+                expect(objs[0]).toMatchObject({
+                    id: obj.id,
+                    iframeUrl: obj.iframeUrl,
+                    type: obj.type,
+                    title: obj.title,
+                    description: obj.description,
+                    meta: obj.meta,
+                    organization: obj.organization,
+                })
+            })
```

Comment:
it's better also check that user can access only their own organization Reports. And can't access another organization owned reports.